src/en/todaymanga/src/eu/kanade/tachiyomi/extension/en/todaymanga/TodayManga.kt

```diff
@@ -0,0 +1,357 @@
+            Pair("Sports", "sports"),
+            Pair("Supernatural", "supernatural"),
+            Pair("Tragedy", "tragedy"),
+            Pair("Vampire", "vampire"),
+            Pair("Webtoons", "webtoons"),
+        ),
+    )
+
+    open class UriPartFilter(displayName: String, private val vals: Array<Pair<String, String>>) :
+        Filter.Select<String>(displayName, vals.map { it.first }.toTypedArray()) {
+        fun toUriPart() = vals[state].second
+    }
+
+    // =========================== Manga Details ============================
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        with(document.selectFirst(".serie")!!) {
+            title = selectFirst("h1")!!.text()
+            thumbnail_url = selectFirst("img")!!.imgAttr()
+            genre = select(".serie-info-head .tags > .tag-item").joinToString { it.text() }
+            author = select(".authors a").joinToString { it.text() }
+            status = selectFirst("li:contains(status) span").parseStatus()
+        }
+
+        description = buildString {
+            val summary = document.selectFirst(".serie-summary")!!
+            summary.childNodes().forEach { node ->
+                if (node is TextNode) {
+                    append(node.text())
+                }
+                if (node.nodeName() == "br") {
+                    appendLine()
+                }
+            }
+            summary.selectFirst("div[style]")?.also {
+                append("\n\n")
+                append(it.text())
+            }
+        }.trim()
+    }
+
+    private fun Element?.parseStatus(): Int = when (this?.text()?.lowercase()) {
+        "complete" -> SManga.COMPLETED
+        "on going" -> SManga.ONGOING
+        else -> SManga.UNKNOWN
+    }
+
+    // ============================== Chapters ==============================
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val chapterHeaders = headersBuilder().apply {
+            add("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8")
+            set("Referer", baseUrl + manga.url)
+        }.build()
+        return GET("$baseUrl${manga.url}/chapter-list", chapterHeaders)
+    }
+
+    override fun chapterListSelector() = "ul.chapters-list > li"
+
+    override fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
+        with(element.selectFirst("a")!!) {
+            name = text()
+            setUrlWithoutDomain(attr("abs:href"))
+        }
+
+        val dateText = element.selectFirst(".subtitle")?.text()
+        date_upload = if (dateText == null) {
+            0L
+        } else if (dateText.contains("ago")) {
+            dateText.parseRelativeDate()
+        } else {
+            parseDate(dateText)
+        }
+    }
+
+    private fun parseDate(dateStr: String): Long {
+        return try {
+            dateFormat.parse(dateStr)!!.time
+        } catch (_: Throwable) {
```

Comment:
```suggestion
        } catch (_: ParseException) {
```

[DateFormat#parse(String)](https://docs.oracle.com/javase/8/docs/api/java/text/DateFormat.html#parse-java.lang.String-) doesn't return null, so you only need to catch [ParseException](https://docs.oracle.com/javase/8/docs/api/java/text/ParseException.html). Catching too wide may suppress other issues.